src/config/src/Node.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <config/Node.hpp>
+
+namespace genny {
+
+void Node::appendKey(std::ostringstream& out) const {
+    if (_parent) {
```

Comment:
It seems weird to me that `appendKey()` is also responsible for appending the parent's key and not just its own. It makes me think a "build path" kind of name is more appropriate.